mkfs/mkfs.c

```diff
@@ -47,8 +47,10 @@ static void exfat_setup_boot_sector(struct pbr *ppbr,
 	pbsx->root_cluster = cpu_to_le32(finfo.root_start_clu);
 	pbsx->vol_serial = cpu_to_le32(1234);
 	pbsx->vol_flags = 0;
-	pbsx->sect_size_bits = bd->sector_size_bits;
-	pbsx->sect_per_clus_bits = log2(ui->cluster_size / bd->sector_size);
+	pbsx->sect_size_bits = 0;
+	/* Compute base 2 logarithm of ui->cluster_size / bd->sector_size */
+	for (i = ui->cluster_size / bd->sector_size; i > 1; i /= 2)
+		pbsx->sect_size_bits++;
```

Comment:
instead of sect_size_bits, sect_per_clus_bits should be calculated.

Reply:
Wow, I don't know how I made that mistake, sorry I'll fix that.